src/EditorFeatures/CSharp/NavigationBar/CSharpNavigationBarItemService.cs

```diff
@@ -53,44 +35,47 @@ public CSharpNavigationBarItemService()
 
         public override async Task<IList<NavigationBarItem>> GetItemsAsync(Document document, CancellationToken cancellationToken)
         {
-            var typesInFile = await GetTypesInFileAsync(document, cancellationToken).ConfigureAwait(false);
+            var documentSymbolsSerivce = document.GetRequiredLanguageService<IDocumentSymbolsService>();
+            var membersInDocument = await documentSymbolsSerivce.GetSymbolsInDocumentAsync(document, DocumentSymbolsOptions.TypesAndMethodsOnly, cancellationToken).ConfigureAwait(false);
+            if (membersInDocument.IsEmpty || cancellationToken.IsCancellationRequested)
```

Comment:
don't really need to check cancellation in this fashion.  generally only need to do `.ThrowIfCancellationRequested` at hte start of methods or loops.